### DISCUSSION TITLE
Allow Cortana Button Functionality to be Togglable

### DISCUSSION THREAD

**Discussion by DavidBerdik** [valinet/ExplorerPatcher#255] [Ideas]
Hello ExplorerPatcher Community,

As of 22000.282.31.4, clicking on the Cortana icon while using the Windows 10 taskbar will open the widgets panel. This is a very welcome change since previous builds did not provide a button to access the widgets and you were therefore stuck using the `WINDOWS KEY + W` keyboard shortcut to access them, however, I am sure there are some users who would prefer for the Cortana icon to work as a Cortana shortcut as it did on Windows 10 and did using previous versions of ExplorerPatcher. To that end, could an additional configuration option be added to allow for users to toggle what the Cortana icon does?

**Comment by valinet** [comment]
Yeah... implemented. Configurable as an option in the GUI:

![image](https://user-images.githubusercontent.com/6503598/139757163-88211ee3-5e8f-4534-8003-5f0484e13f36.png)

If one enables it from here (taskbar context menu), it defaults to opening Cortana:

![image](https://user-images.githubusercontent.com/6503598/139757740-7fda7e77-bf4f-4af2-8ec3-ab24b20fd26d.png)

If one enables it from here (Taskbar settings), it defaults to opening Widgets:

![image](https://user-images.githubusercontent.com/6503598/139757848-8c64661f-4a06-4483-872a-c310c97a3242.png)

One can of course change it to their liking in the GUI afterwards. I think it is okay like this for the moment.

Thanks

**Comment by DavidBerdik** [comment]
Wow! That was an impressively fast turnaround time! Thank you! 😁

**Reply by valinet** [reply]
You’re welcome. I am glad I was able to pull it off so quick too.
